src/motioncore/algorithm/algorithm_description.cpp

```diff
@@ -126,8 +126,13 @@ AlgorithmDescription AlgorithmDescription::FromBristol(std::ifstream& stream) {
     } else if (type == std::string("MUX")) {
       assert(line_vector.size() == 7);
       primitive_operation.type = PrimitiveOperationType::kMux;
-      primitive_operation.parent_a = std::stoull(line_vector.at(2));
-      primitive_operation.parent_b = std::stoull(line_vector.at(3));
+
+      // ! error correction: switch parent_b and parent_a for MUX gate
+      // primitive_operation.parent_a = std::stoull(line_vector.at(2));
+      // primitive_operation.parent_b = std::stoull(line_vector.at(3));
+      primitive_operation.parent_b = std::stoull(line_vector.at(2));
```

Comment:
Where did you encounter problems with that? Was this due to a different layout of inputs in some format, HyCC? If so, this would rather require a change in the parser's logic than in the `AlgorithmDescription`s logic. Does that make sense?

Reply:
Commit is reverted and change parser's logic

Reply:
Is this part of code actually used anywhere? You changed the logic in `AlgorithmDescription` but not the tests, and they still run. So it isn't used anywhere currently?

Please also comment other tests back in. It's important to make sure that *all* tests run before basically any MR can be merged.

Reply:
Hi, I have reverted the commit (the [code after revert](https://github.com/liangzhao-darmstadt/MOTION/blob/ed40d01c3d36ebacb1ac3add70e8120445338b4e/src/motioncore/algorithm/algorithm_description.cpp#L126) is actually the same as [encryptogroup/MOTION](https://github.com/encryptogroup/MOTION/blob/ed40d01c3d36ebacb1ac3add70e8120445338b4e/src/motioncore/algorithm/algorithm_description.cpp#L126)). 
This fix for mux gate is essential for floating-point arithmetic (Boolean GMW).
It would be better to commit this fix when making pull requests for floating-point arithmetic operations.

Reply:
sounds reasonable